### DISCUSSION TITLE
Connecting a UI

### DISCUSSION THREAD

**Discussion by trya2l** [rwf2/Rocket#2781] [Questions] [answered]
Hi guys,

It is really a newbie question, but I need to ask it because I am confused about how to start with rocket.

Do you have advices to build a UI using rocket ? Are there some facilities (some plugins) to build a crud admin interface, for example like in django or symfony ?

Do you know about some resources that could help me learn well ?

**Comment by calteran** [comment, marked as answer]
Hi @trya2l , my impression/experience with Rocket is that it's not really a frontend framework.  I'm using it to build a large API, and it's fantastic for that, but you'll need a different framework to generate markup, styling and scripting for your UI.  Personally, I'm using Next.js for that, but there are plenty of other solutions out there too, and I hear there are some decent ones in Rust coming along.

If you want to run your UI code from Rocket directly, you'll either need to hard-code it into Rocket's route functions or call your UI framework from those functions.

FWIW, the above is just my opinion & based on ~6 months experience with Rocket and I'm happy if someone with more perspective or insight can tell us that I'm wrong.

**Reply by the10thWiz** [reply]
You are absolutely correct. Rocket is only a backend framework, for building a Web Server. There are fairings for pre-compiling SASS to CSS, but Rocket doesn't (and likely never will) support anything akin to React/JSX.

**Reply by trya2l** [reply]
Thanks guys, this clarified some points.
